shell/browser/api/electron_api_base_window.cc

```diff
@@ -783,6 +783,25 @@ void BaseWindow::RemoveBrowserView(v8::Local<v8::Value> value) {
   }
 }
 
+void BaseWindow::SetTopBrowserView(v8::Local<v8::Value> value) {
+  gin::Handle<BrowserView> browser_view;
+  if (value->IsObject() &&
+      gin::ConvertFromV8(isolate(), value, &browser_view)) {
+    if (!browser_view->web_contents())
+      return;
+    auto get_that_view = browser_views_.find(browser_view->ID());
+    if (get_that_view == browser_views_.end()) {
```

Comment:
We need to be careful here to ensure that the `BrowserView` which is being raised on a given `BrowserWindow` is not already attached to another window. If for example I call:

```js
  const win1 = new BrowserWindow()
  const win2 = new BrowserWindow()

  const view = new BrowserView()
  win1.addBrowserView(view)
  view.setBounds({ x: 0, y: 0, width: 100, height: 100 })
  win2.addBrowserView(view)
```

and then later call 
```js
win1.setTopBrowserView(view)
```

what behavior would a user reasonably expect to happen? Would this reparent the existing `BrowserView`? (Also see https://github.com/electron/electron/pull/27000 for related behavior)

I would say here that since `setTopBrowserView` makes the implicit assumption that the view _already_ exists on the parent window, that we should not be reparenting it in the event of the above - we should likely instead raise an exception or otherwise act in a way that least surprises the user.

Reply:
Good call. I assumed it was safe because of how `addBrowserView()` was written. Seeing your change makes me think we should throw (rather than copying the fix).